main.ts

```diff
@@ -8,153 +8,161 @@ import fs = require("fs");
 const maxUploadFiles: number = 3; // no more than 3 files can be uploaded at a time
 const maxRetries: number = 3; // max number of uploads to retry
 
+async function upload_step_init(dt_upload_api_key: string): Promise<Response> {
+  return await fetch(
+    "https://api.securetheorem.com/uploadapi/v1/upload_init",
+    {
+      "headers": {
+        "Accept": "application/json",
+        "Authorization": "APIKey " + dt_upload_api_key,
+        "Content-Type": "application/json",
+      },
+      "method": "POST",
+    },
+  );
+}
+
 async function run() {
-  try {
```

Comment:
This makes the change a little longer but it does unnest this function by one level. It could be simplified further but since this is not the scope I tried to limit the refactoring here.